plugins/experimental/slice/response.cc

```diff
@@ -68,10 +68,13 @@ string502()
     bodystr.append("</body>\n");
     bodystr.append("</html>\n");
 
+    char hverstr[64];
+    int const hlen =
+      snprintf(hverstr, sizeof(hverstr), "HTTP/%d.%d 502 Bad Gateway\r\n", TS_HTTP_MAJOR(httpver), TS_HTTP_MINOR(httpver));
+    msg.append(hverstr, hlen);
+
```

Comment:
I assume this is the heart of your 502 fixes for this PR.  Much of the rest looks like more general cleanup.

Reply:
Yes sort of.  The original PR was much more aggressive about trying to use the state machine but attempts failed and this cleanup was what was left over.